pkg/workloadmeta/types.go

```diff
@@ -546,16 +555,40 @@ type CollectorEvent struct {
 	Entity Entity
 }
 
-// Event is an event generated by the metadata store, to be broadcasted to
-// subscribers.
+// Event represents a change to an entity.
 type Event struct {
-	Type    EventType
+	// Type gives the type of this event.
+	//
+	// When Type is EventTypeSet, this represents an added or updated entity.
+	// Multiple set events may be sent for a single entity.
+	//
+	// When Type is EventTypeUnset, this represents either a removed entity or
```

Comment:
When a source removes the information for an entity, we decide whether or not to send a Set or Unset based on filters. If the subscriber has a filter, and it filters out the info from the remaining sources, we send an `Unset` (as from that subscriber's point of view, their entity is gone). If there are remaining sources, it's just a simple update, so we send a `Set`. So it's not possible for a subscriber to receive more than one Unset per entity (unless my code's broken, of course 😅).

I agree it's very hard to make that out from reading the code, and that's one reason why #10331 changes this significantly.

Reply:
Ah, should this 
https://github.com/DataDog/datadog-agent/blob/e8bae0f8d181e9e84653a33f259dd5135774a16f/pkg/workloadmeta/store.go#L435-L438
possibly be EventTypeSet, then?  That answers one of my questions :)